core/controllers/suggestion.py

```diff
@@ -344,8 +352,24 @@ def get(self, target_type, suggestion_type):
         """
         self._require_valid_suggestion_and_target_types(
             target_type, suggestion_type)
+        topic_name = self.request.get('topic_name', None)
```

Comment:
Any reason not to pass the topic ID instead? Seems like a more logical thing to use as the identifier.

Reply:
I believe either is fine? topic_name just adds more readability IMO for anyone who is tracing the code in the backend and trying to understand the feature. But I can replace with topicId if you think that's better.

Reply:
topic ID is actually the way to go for retrieval -- you want a unique ID and that's how we typically identify entities. It also reduces a fetch. 

(Technically, the "names" of two different things can be the same.)

Reply:
Oh, okay. I have been using this as reference (https://github.com/bhavukJain/oppia/blob/4917644d93c7dd3785d83de8d4e94294458eb281/core/controllers/contributor_dashboard.py#L102) and I believe, while saving the topic to datastore, the datastore won't allow having same ID for two different topics? 

Since this is passed from front-end, also adding @jimbyo to get this thoughts. @jimbyo can we get topic id directly from the front-end?

Reply:
For topics, you're right -- those are actually a special case because we support fetching by topic name in the URLs (unlike for other entities).

So if topic ID is not available in the frontend, then I think it's ok to go with topic name as a backup. But let's see if topic ID is available first because that is more canonical.

Reply:
I don't see a quick way to do it, For now lets keep it as topic_name however.  But I'll investigate more tommorow morning and will update you all again

Reply:
Even if we end passing a topic id, we might need to make an additional datastore call in the frontend.

Reply:
OK with me -- if the topic ID isn't already available in the frontend then let's not go through additional contortions. (I had assumed that it was.)

Reply:
Kept it same as is was before.